web/src/main/java/org/openmrs/web/filter/initialization/InitializationFilter.java

```diff
@@ -1936,14 +1957,39 @@ private static boolean goBack(HttpServletRequest httpRequest) {
 		return "Back".equals(httpRequest.getParameter("back"))
 			|| (httpRequest.getParameter("back.x") != null && httpRequest.getParameter("back.y") != null);
 	}
-	
+
+	/**
+	 * Resolves and merges initialization properties from multiple sources to configure the setup wizard.
+	 *
+	 * @return a {@link Properties} object containing the resolved initialization configuration.
+	 */
+	protected Properties resolveInitializationProperties() {
+		Properties merged = new Properties();
+		System.getProperties().forEach((key, value) -> merged.putIfAbsent(key.toString(), value.toString()));
+		getEnvironmentVariables().forEach((key, value) -> merged.putIfAbsent(normalizeEnvVariableKey(key), value));
+		Properties installScript = getInstallationScript();
+		installScript.forEach((key, value) -> merged.putIfAbsent(key.toString(), value.toString()));
+		return merged;
+	}
+
+	private String normalizeEnvVariableKey(String envVarKey) {
+		if (NON_NORMALIZED_KEYS.contains(envVarKey)) {
+			return envVarKey.toLowerCase();
+		}
+		return envVarKey.toLowerCase().replace('_', '.');
+	}
+
+	protected Map<String, String> getEnvironmentVariables() {
+		return System.getenv();
```

Comment:
Are you only checking for OS env variables? We also want to combine them with System.getProperties() that are java environment variables. The order should be:
1. Java env variables
2. OS env variables
3. properties files
Meaning 1. overwrites 2 and 3.